### DISCUSSION TITLE
How to run the container using --network on docker?

### DISCUSSION THREAD

**Discussion by andriansandi** [engageub/InternetIncome#66] [General]
Hello,

Do the script can run using adding parameter --network ? Since I have more than one IP Public on my Server

**Comment by engageub** [comment]
Hello,
For multiple IP,s please use the test branch. The changes have been made to support public IPs but not yet tested.
You have to add your public IP address in multiips.txt file and set USE_MULTI_IP to true in properties.conf file.

https://github.com/engageub/InternetIncome/tree/test


Thank you

**Reply by engageub** [reply]
If you wish to manually set network in main branch, you will have to set that in line 181 of the code in main branch.
After downloading it from main branch you can edit internetIncome.sh file. 

https://github.com/engageub/InternetIncome/blob/main/internetIncome.sh#L181

Thank you

**Reply by andriansandi** [reply]
I think we need to mention about iptables NAT for the IP Addresses?

**Reply by engageub** [reply]
Yes, iptables configuration is mentioned in test branch but since you are were looking for --network addition I mentioned about the main branch.
Please confirm if test branch is working fine with multiple IP address. 

Thank you

**Reply by andriansandi** [reply]
no working for ebesucher and adnade, but it's fine using the gaganode

**Comment by engageub** [comment]
> no working for ebesucher and adnade, but it's fine using the gaganode

Are you saying that multi IP is working with gaganode. Do you see different public IP address in gaganode dashboard.
Adnade and ebesucher may not work with 127.0.0.1 as they will be on different network interface while using public network. Please try to use your Public IP and port number as mentioned in ebesucher.txt.


Thank you

**Reply by andriansandi** [reply]
the ebesucher.txt still using 127.0.0.1:300 don't know why

**Comment by engageub** [comment]
Please check your earnings for adnade and ebesucher if there are no errors
while starting the
script or in docker container logs.

Thank you

**Comment by engageub** [comment]
Hi,
Its just a reference to know on what port it is running. Its not required
to view the container. Please check your earnings.

Thank  you

**Reply by andriansandi** [reply]
![image](https://github.com/engageub/InternetIncome/assets/596269/cd3da739-e275-4bca-91f2-fafec1761268)

Is that normal for the multiple IPs?

**Comment by engageub** [comment]
Hi,
I have not tested multiple IPs as I don't have the setup.
You can look into docker container logs by enabling logs option in properties.conf file.
Please confirm if you are seeing the IP address in gaganode dashboard. 
If you are seeing your public IP address in gaganode dashboard, then the code is working fine. 

As mentioned 127.0.0.1 is reference that its running on your localhost. But in your case you are using different network with different interface. 
Please use http://YOUR_PUBLIC_IP:PORT to see if its accessible.

You can access the docker logs by running the following command after enabling logs in properties.conf

```
sudo docker container logs YOUR_CONTAINER_NAME
```

**Reply by engageub** [reply]
From the code, I see ports are not opened for multiple IPs. But that should not be a problem for earnings.
The code will be updated soon with open ports for multiple IP address. 
You have to confirm to me that you are seeing different public IP address in gaganode dashboard.

Thank you

**Comment by engageub** [comment]
Code has been updated to open ports when multiple ip are used.
Could you please download it again.
Please make sure you delete the previous containers using --delete option

https://github.com/engageub/InternetIncome/tree/test

Thank you

**Comment by andriansandi** [comment]
Why for the adnade when using multiple IP will be using PTP links instead of surfing ?

**Comment by engageub** [comment]
For test branch you have to use chrome for adnade for surfing.
Please change in properties.conf.
It looks like you changed the default values. 

I am asking you again.
Do you see your public Ips in gaganode?
Thank you

**Comment by andriansandi** [comment]
I just crashed my server because running 8 IPs with 3 containers: adnade, ebesucher and gaganode lol. My server only 8GB. 

Btw, I've got the 2nd server with 10 IPs and trying to running ebesucher and adnade only with 5 IPs. The ebesucher seems working but the adnade stuck on Google Recaptcha

![image](https://github.com/engageub/InternetIncome/assets/596269/81342f3d-74c3-48a7-b46b-15c60832bc0a)

**Comment by engageub** [comment]
Hi,
Did you check your public IP address in the browser or in gaganode dashboard. Are they using public IP address or your VPS IP address.

Thank you

**Reply by andriansandi** [reply]
Hi, 

All the containers for the ebesucher and adnade have different public IP

**Reply by engageub** [reply]
Thank you for the confirmation. Adnade restarts every two hours and Ebesucher restarts every one hour.
PTP was kept in firefox and surfbar for chrome in test branch for testing.
The recaptcha is solved by hektcaptcha extension automatically. May be you have to wait for the recaptcha to expire after 2minutes so that it opens the new recaptcha again. It keeps retrying till the recaptcha window is open again. 

Thank you

**Reply by andriansandi** [reply]
Okay, I will wait for that. Do you think best if using the proxy instead of the multiple IPs?

**Reply by engageub** [reply]
Actually with multiple IPs, you will not have any issues as all the protocols required are available to communicate. Its a separate network.
Mysterium was not working with proxies. But you can try that using multiple IPs. Earnings are good for Mysterium, sometimes users earn about $100 per month per IP address. You can also try honeygain which pays good for residential proxies.
May I know from where I can get a host with multiple IP address.

Thank you

**Reply by andriansandi** [reply]
I think the issue on chrome for the recaptcha solver, I tried another VMs using firefox + proxy working well. 

For the Residential Hosting that I've tried:
1. soladrive.com
2. cloudfitters.com
3. gokturhost (turkey based, cheap but bad support lol)

**Reply by engageub** [reply]
Thank you for the response. Made some modifications for chromebrowser and is better now as hardware acceleration is not compatible with all systems.
Please delete using --delete and delete the folders then download the code from test branch again.

Thank you

**Comment by engageub** [comment]
Test branch has been updated to use surfing in firefox for adnade containers.
Please download it again. 
Thank you

**Reply by engageub** [reply]
Please see the discussion here to use custom network.

https://github.com/engageub/InternetIncome/discussions/50

**Comment by engageub** [comment]
Hello, 
Code for multi IP has been updated recently.
You may download the latest version. 
https://github.com/engageub/InternetIncome/tree/test

Thank you
